src/bindgen/mangle.rs

```diff
@@ -57,11 +59,26 @@ impl<'a> Mangler<'a> {
         match *ty {
             Type::Path(ref generic) => {
                 let sub_path =
-                    Mangler::new(generic.export_name(), generic.generics(), last, Some(self.mangle_separator)).mangle();
+                    Mangler::new(
+                        generic.export_name(),
+                        generic.generics(),
+                        last,
+                        Some(self.mangle_separator),
+                        self.capitalize_primitives).mangle();
+
                 self.output.push_str(&sub_path);
             }
             Type::Primitive(ref primitive) => {
-                self.output.push_str(primitive.to_repr_rust());
+                let mut primitive_string = primitive.to_repr_rust().to_owned();
```

Comment:
Can we use `Cow::Borrowed`, and only allocate if the new option is used?

Reply:
Done

Reply:
Rebased to master and ran `rustfmt`